python/jupytergis_qgis/jupytergis_qgis/qgis_loader.py

```diff
@@ -477,6 +485,65 @@ def build_uri(parameters: dict[str, str], source_type: str) -> str | None:
 
             renderer.setStyles(parsed_styles)
 
+    if layer_type == "VectorLayer" and source_type == "GeoJSONSource":
+        source_parameters = source.get("parameters", {})
+        uri = build_uri(source_parameters, "GeoJSONSource")
+        if not uri:
+            logs["warnings"].append(
+                f"Layer {layer_id} not exported: invalid GeoJSON source."
+            )
+            return
+
+        map_layer = QgsVectorLayer(uri, layer_name, "ogr")
+        if not map_layer.isValid():
+            logs["warnings"].append(
+                f"Layer {layer_id} not exported: failed to load GeoJSON source."
+            )
+            return
+
+        geometry_type = map_layer.geometryType()
+        layer_params = layer.get("parameters", {})
+        print(f"Geometry Type: {geometry_type}", layer_params)
```

Comment:
```suggestion
```

Reply:
Still testing, Will make sure to remove these before final iteration :)